blackbox/docs/general/builtins/scalar.rst

```diff
@@ -1777,6 +1777,63 @@ Example::
     SELECT 1 row in set (... sec)
 
 
+
+Special Functions
+=================
+
+.. _ignore3vl:
+
+``ignore3vl(boolean)``
+----------------------
+
+The ``ignore3vl`` function operates on a boolean argument and eliminates the
+`3-valued logic`_ on the whole tree of operators beneath it. More specifically,
+``FALSE`` is evaluated to ``FALSE``, ``TRUE`` to ``TRUE`` and ``NULL`` to
+``FALSE``.
+
+Returns: ``boolean``
+
+.. NOTE::
+
+    The main usage of the ``ignore3vl`` function is in the ``WHERE`` clause
+    when a ``NOT`` operator is involved. Such filtering, with
+    `3-valued logic`_, cannot be translated to an optimized query in the
+    internal storage engine, and therefore can result into slow performance.
+    E.g.::
+
+      SELECT * FROM t
+      WHERE NOT 5 = ANY(t.long_array_col)
+
+    If we can ignore the `3-valued logic`_, we can write the query as::
+
+      SELECT * FROM t
+      WHERE NOT IGNORE3VL(5 = ANY(t.long_array_col)
+
+    which will yield better performance (in execution time) than before.
+
+    .. WARNING::
+
+      If there are NULL values in the `long_array_col`, in the case that
+      `5 = ANY(t.long_array_col)` evaluates to ``NULL``, without the
+      ``ignore3vl``, it would be evaluated as ``NOT NULL`` => ``NULL``,
+      resulting to zero matched rows. With the ``IGNORE3VL`` in place it will
+      be evaluated as ``NOT FALSE`` => ``TRUE`` resulting to all rows matching
+      the filter.
+
+Synopsis::
+
+    ignore3vl(boolean)
+
+Example::
+
+    cr> SELECT ignore3vl(true) as v1, ignore3vl(false) as v2, ignore3vl(null) as v3 FROM sys.cluster
+    +----------------------+
+    |   v1 |    v2 |    v3 |
+    +----------------------+
+    | true | false | false |
+    +----------------------+
+    SELECT 1 row in set (... sec)
```

Comment:
can we also give an example with an array?
i guess this would be really helpful as this is the main use case.

Reply:
Isn't that enough: https://github.com/crate/crate/pull/7422/files#diff-c1428c761c17d1bb4ed67b8e27361a96R1804 ?

You mean a full example with crash and the results?

Reply:
no, i'd just have an example containing arrays.
something like that:

```
SELECT * FROM t WHERE NOT IGNORE3VL(5 = ANY([0, 1, null]));
```

plus the output.